lib/src/view/coordinate_training/coordinate_training_screen.dart

```diff
@@ -47,17 +47,31 @@ class _Body extends ConsumerStatefulWidget {
 class _BodyState extends ConsumerState<_Body> {
   late Side orientation;
 
+  late bool computeRandomOrientation;
```

Comment:
This new `boolean` and the `_maybeSetOrientation` makes the code harder to read and understand.

This is a sign that now this logic shouldn't be in the widget itself, as it is not an ephemeral state.

Adding the `orientation` as a property of the `CoordinateTrainingState`, and recompute the orientation at each new session if `random` pref is chosen, would be much clearer.

Reply:
Even if I move this to the state, I'll still need the boolean but what I can do instead is to choose the random side when the user either aborts or ends a training.

This way the boolean would not be needed and I think it is a better experience for the user to have the orientation before starting the training. This will differ from the website but I believe it is better for the user and easier to code.

I think it still remains a good idea to move this to the coordinate training controller.